arch/xtensa/core/mpu.c

```diff
@@ -595,10 +599,7 @@ static int mpu_map_region_add(struct xtensa_mpu_map *map,
 		xtensa_mpu_entry_attributes_set(&entries[idx], access_rights, memory_type);
 	}
 
-	if (first_idx != NULL) {
-		*first_idx = first_enabled_idx;
-	}
-
+	*first_idx = first_enabled_idx;
```

Comment:
As mentioned in comment above that `first_idx == NULL` is a common use case, please do not remove the `if`.

Reply:
fixed